packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -330,6 +332,23 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
     }
   }
 
+  private _setTargetWidthAsync() {
+    const targetAsHtmlElement = this._target as HTMLElement;
+
+    // If component doesn't care about target's width, then this is a no-op.
+    if ((!this.props.useTargetWidth && !this.props.useTargetAsMinWidth) || !targetAsHtmlElement) {
+      return;
+    }
+
+    this._async.requestAnimationFrame(() => {
+      const targetBoundingRect = targetAsHtmlElement.getBoundingClientRect();
+
+      this.setState({
+        targetWidth: targetBoundingRect.width - 2 /* Need to account for border width */
```

Comment:
>targetWidth [](start = 8, length = 11)

Since this async, how do we make sure this value is set when the render API is called?

Reply:
It gets called in `componentWillMount`, so the value should be available in the `render` function. We'll have this issue either way since @joschect said even reading `offsetWidth` should be moved to an async function. Can someone more familiar with this pattern in OUFR help solve the problem? Do you have any suggestions, @manishgarg1?

Reply:
Or if you have a better way to solve it, feel free to take it over. The issue has been open since November with no activity and we need a fix. Thanks!

Reply:
It's okay if the value is not set at first render, although it may cause apparent resizing. You would need to provide a default value though.